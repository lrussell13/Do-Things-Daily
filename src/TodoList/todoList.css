.todo {
    padding: 15px;
}
