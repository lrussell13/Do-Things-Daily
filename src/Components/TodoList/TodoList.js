import React from 'react';
import Todo from './Todo/Todo';
import ProgressBar from './ProgressBar/ProgressBar';
import './todoList.css';


function TodoList(props) {
    return (
        <div>
          <h1>Today</h1>
          <ProgressBar todos={props.todos}/>
          <hr id="main-hr"/>
          {props.todos.map((todo, i) => (
          <Todo key={String(todo.id)} {...todo} deleteTodo={props.deleteTodo} checkTodoNow={props.checkTodo}/>
        ))}
        </div>
    )
}


export default TodoList;