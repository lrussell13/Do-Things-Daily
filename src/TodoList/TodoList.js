import React from 'react';
import Todo from './Todo/Todo';
import ProgressBar from './ProgressBar/ProgressBar';
import './todoList.css';
import { todoServices } from '../TodoServices';


class TodoList extends React.Component {
  state = {
    name: '',
    todos: [],
    id: 4
  }

  checkTodo = id => {

    function find(element) {
      return element.id === id
    }
    
    const todosNew = this.state.todos;
    const index = todosNew.findIndex(find);
    const checked = !todosNew[index].checked;
    const data = {"checked": checked};

    todoServices.checkTodos(data, id)
    .then(() => todoServices.fetchTodos()
    .then(todos => this.setState({todos})));
  }

  componentWillMount(){
    todoServices.fetchTodos().then(todos => this.setState({todos}))
  }

  render(){
    return (
        <div>
          <h1>Today</h1>
          <ProgressBar todos={this.state.todos}/>
          <hr id="main-hr"/>
          {this.state.todos.map((todo, i) => (
          <Todo key={String(todo.id)} {...todo} deleteTodo={this.deleteTodo} checkTodoNow={this.checkTodo}/>
        ))}
        </div>
    )
  }
}

export default TodoList;