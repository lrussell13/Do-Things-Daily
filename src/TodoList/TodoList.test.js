import React from 'react';
import ReactDOM from 'react-dom';
import TodoList from './TodoList';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TodoList todos={[
        {
          "id": 1,
          "name": "Code",
          "checked": false
        },
        {
          "name": "Read",
          "checked": false,
          "id": 2
        },
        {
          "name": "Brush teeth",
          "checked": false,
          "id": 3
        },
        {
          "name": "Something",
          "checked": false,
          "id": 4
        }
      ]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<TodoList todos={[
            {
              "id": 1,
              "name": "Code",
              "checked": false
            },
            {
              "name": "Read",
              "checked": false,
              "id": 2
            },
            {
              "name": "Brush teeth",
              "checked": false,
              "id": 3
            },
            {
              "name": "Something",
              "checked": false,
              "id": 4
            }
          ]}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();  
});