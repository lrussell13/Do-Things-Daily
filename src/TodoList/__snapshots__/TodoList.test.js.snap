// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <h1>
    Today
  </h1>
  <div
    className="progress-container"
  >
    <div
      className="progress-bar"
    />
  </div>
  <hr
    id="main-hr"
  />
  <div
    className="todo"
  >
    <div
      className="todo-items"
    >
      <label
        className="container"
      >
        Code
        <input
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <hr />
  </div>
  <div
    className="todo"
  >
    <div
      className="todo-items"
    >
      <label
        className="container"
      >
        Read
        <input
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <hr />
  </div>
  <div
    className="todo"
  >
    <div
      className="todo-items"
    >
      <label
        className="container"
      >
        Brush teeth
        <input
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <hr />
  </div>
  <div
    className="todo"
  >
    <div
      className="todo-items"
    >
      <label
        className="container"
      >
        Something
        <input
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <hr />
  </div>
</div>
`;
